Docs/dev/how_to_add_a_new_sensor.md

```diff
@@ -0,0 +1,569 @@
+Vehicle too close: vehicle.mercedes-benz.coupe
+```
+
+That's it, we have a new sensor working!
+
+- - -
+
+## Appendix: Reusing buffers
+
+In order to optimize memory usage, we can use the fact that each sensor sends
+buffers of similar size; in particularly, in the case of cameras, the size of
+the image is constant during execution. In those cases, we can save a lot by
+reusing the allocated memory between frames.
+
+Each stream contains a _buffer pool_ that can be used to avoid unnecessary
+memory allocations. Remember that each sensor has a stream associated thus each
+sensor has its own buffer pool.
+
+Use the following to retrieve a buffer from the pool
+
+```cpp
+auto Buffer = Stream.PopBufferFromPool();
+```
+
+If the pool is empty, it returns an empty buffer, i.e. a buffer with no memory
+allocated. In that case, when you resize the buffer new memory will be
+allocated. This will happen a few times during the first frames. However, if a
+buffer was retrieved from the pool, its memory will go back to the pool once the
+buffer goes out of the scope. Next time you get another buffer from the pool,
+it'll contain the allocated piece of memory from the previous buffer. As you can
+see, a buffer object acts actually as an smart pointer to a contiguous piece of
+raw memory. As long as you don't request more memory than the currently
+allocated, the buffer reuses the memory. If you request more, then it'll have to
+delete the current memory and allocate a bigger chunk.
+
+The following snippet illustrates how buffers work
+
+```cpp
+Buffer buffer;
+buffer.reset(1024u); // (size 1024 bytes, capacity 1024 bytes) -> allocates
+buffer.reset(512u);  // (size  512 bytes, capacity 1024 bytes)
+buffer.reset(2048u); // (size 2048 bytes, capacity 2048 bytes) -> allocates
+```
+
+## Appendix: Sending data asynchronously
+
+Some sensors may require to send data asynchronously, either for performance or
+because the data is generated in a different, for instance, camera sensors send
+the images from the render thread.
+
+Using the data stream asynchronously is perfectly fine, as long as the stream
+itself is created in the game thread. For instance
+
+```cpp
+void MySensor::Tick(float DeltaSeconds)
+{
+  Super::Tick(DeltaSeconds);
+
+  auto Stream = GetDataStream(*this);
+
+  std::async(std::launch::async, [Stream=std::move(Stream)]() {
+    auto Data = ComputeData();
+    Stream.Send(*this, Data);
+  });
+}
+```
+
+## Appendix: Client-side sensors
+
+Some sensors do not require the simulator to do their measurements, those
+sensors may run completely in the client-side freeing the simulator from extra
+computations. Examples of such sensors are the _GNSS_ and the _LaneInvasion_
+sensors.
+
+The usual approach is to create a "dummy" sensor in the server-side, just so the
+simulator is aware that such actor exists. However, this dummy sensor does tick
+nor sends any sort of data. Its counterpart on the client-side however,
```

Comment:
Was the intention here maybe to write
>However, this dummy sensor `doesn't` tick nor sends any sort of data.

?

Reply:
Yes! thanks I'll fix that